### PR TITLE
sys/cpp11-compat: Remove xtimer deps

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

This replaces the `xtimer` calls with explicit `ztimer64` calls, since `xtimer` is somewhat deprecated. 

### Testing procedure

- Green murdock
- I suppose the following `tests/`
```
c11_atomics_cpp_compat
cpp11_condition_variable
cpp11_mutex
cpp11_thread
cpp_ctors
cpp_exclude
cpp_ext
irq_cpp
rmutex_cp
```
- run `compile_and_test_for_board.py` and `compile_like_murdock.py` for the subset of tests.

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Any blockers? @josephnoir originally did this?

File: sys/cpp11-compat/include/riot/condition_variable.hpp [hunk 1]
```diff
@@ -176,12 +177,13 @@ cv_status condition_variable::wait_for(unique_lock<mutex>& lock,
   timeout.seconds = s.count();
   timeout.microseconds
     = (duration_cast<microseconds>(timeout_duration - s)).count();
-  xtimer_now_timex(&before);
-  xtimer_t timer;
-  xtimer_set_wakeup(&timer, timex_uint64(timeout), thread_getpid());
+  before = timex_from_uint64(ztimer64_now(ZTIMER64_USEC));
```
Comment (kfessel) [inline sys/cpp11-compat/include/riot/condition_variable.hpp:180]: Why is duration cast to timex_t (l176 to 179) ? (not to 64bit microseconds directly)

File: tests/cpp11_condition_variable/main.cpp [hunk 2]
```diff
@@ -97,9 +99,9 @@ int main() {
     condition_variable cv;
     timex_t before, after;
     unique_lock<mutex> lk(m);
-    xtimer_now_timex(&before);
+    before = timex_from_uint64(ztimer64_now(ZTIMER64_USEC));
     cv.wait_for(lk, chrono::seconds(timeout));
-    xtimer_now_timex(&after);
+    after = timex_from_uint64(ztimer64_now(ZTIMER64_USEC));
```
Comment (kfessel) [inline tests/cpp11_condition_variable/main.cpp:104]: seems also like unforced use of timex

Comment (MrKevinWeiss) [reply tests/cpp11_condition_variable/main.cpp]: Removed

**Review by MrKevinWeiss — COMMENTED** [review]
Note that there is still a dependency on timex for the type...

File: sys/cpp11-compat/include/riot/condition_variable.hpp [hunk 3]
```diff
@@ -171,20 +172,20 @@ cv_status condition_variable::wait_for(unique_lock<mutex>& lock,
...
+  uint64_t timeout, before, after;
   auto s = duration_cast<seconds>(timeout_duration);
-  timeout.seconds = s.count();
-  timeout.microseconds
-    = (duration_cast<microseconds>(timeout_duration - s)).count();
-  xtimer_now_timex(&before);
-  xtimer_t timer;
-  xtimer_set_wakeup(&timer, timex_uint64(timeout), thread_getpid());
+  timeout = s.count() * US_PER_SEC;
+  timeout += (duration_cast<microseconds>(timeout_duration - s)).count();
```
Comment (kfessel) [inline sys/cpp11-compat/include/riot/condition_variable.hpp:178]:
```suggestion
  timeout = duration_cast<microseconds>(timeout_duration).count();
```
and remove line 176

Comment (MrKevinWeiss) [reply sys/cpp11-compat/include/riot/condition_variable.hpp]: Right, I thought that microseconds overflow at 1 second but I guess that was done as a result of `timex`.  Done.

**Review by kfessel — COMMENTED** [review]
reads good, tested good

please squash,
while doing that please correct the commit messages to "*: Use ztimer64_usec instead of xtimer" as these are ports to ztimer64 (which requires no aquire and release and has the timer (and therefor clock) always running)

**Review by kfessel — APPROVED** [review]

**Comment by MrKevinWeiss** [comment]
Thanks!

**Comment by MrKevinWeiss** [comment]
bors merge

**Comment by MrKevinWeiss** [comment]
Hmm I don't like how it is bigger...

**Comment by kfessel** [comment]
it shrunk for stm32f7 (m7) and stm32f3(m4) and increased for stm32l0 (m0)
```
   text    data     bss     dec     hex filename
  36364     196    2476   39036    987c cpppr/tests/cpp11_thread/bin/nucleo-f767zi/tests_cpp11_thread.elf
  36584     196    2476   39256    9958 master/tests/cpp11_thread/bin/nucleo-f767zi/tests_cpp11_thread.elf
  36072     192    2468   38732    974c cpppr/tests/cpp11_thread/bin/nucleo-l053r8/tests_cpp11_thread.elf
  35908     192    2468   38568    96a8 master/tests/cpp11_thread/bin/nucleo-l053r8/tests_cpp11_thread.elf
  35784     192    2476   38452    9634 cpppr/tests/cpp11_thread/bin/nucleo-f303re/tests_cpp11_thread.elf
  36004     192    2476   38672    9710 master/tests/cpp11_thread/bin/nucleo-f303re/tests_cpp11_thread.elf
```

**Comment by MrKevinWeiss** [comment]
bors merge

**Comment by MrKevinWeiss** [comment]
bors merge

**Pull request merged**